src/Npgsql/TypeMapping/TypeMapper.cs

```diff
@@ -160,6 +165,22 @@ NpgsqlTypeHandler ResolveLong(NpgsqlDbType npgsqlDbType)
                     }
                 }
 
+                // Can't find (or translate) PG data type name by NpgsqlDbType.
```

Comment:
I'm not sure any more why this range and array code are required; the built-in range types (e.g. NpgsqlDbType.TimestampTzRange) and arrays are already hard-coded/handled in GlobalTypeMapper/NpgsqlDbTypeToDataTypeName, no? If that's true, we should be able to remove the logic below for range/arrays without adding this bit.

Basically the logic is that NpgsqlDbType is closed, since it's an enum - it won't evolve; and the mapping between NpgsqlDbType and data types is fixed as well, no need to have resolvers involved in that.

What am I missing?

Reply:
Should have been removed after rebase...

Reply:
I think this still needs to be done

Reply:
Ah, I see what you mean. No, we still need this because `BuiltinResolver.ResolveByDataTypeName` doesn't handle arrays or ranges.

Reply:
Right, thanks for reminding me :)